barretenberg/cpp/src/barretenberg/vm/tests/AvmMini_execution.test.cpp

```diff
@@ -10,6 +10,24 @@
 #include <string>
 #include <utility>
 
+namespace {
+void gen_proof_and_validate(std::vector<uint8_t> const& bytecode,
+                            std::vector<Row>&& trace,
+                            std::vector<FF> const& calldata)
+{
+    auto circuit_builder = AvmMiniCircuitBuilder();
+    circuit_builder.set_trace(std::move(trace));
+    EXPECT_TRUE(circuit_builder.check_circuit());
```

Comment:
Just a general GTest comment (just in case): `EXPECT_` should only be used to check what the test case if testing, and not be used to check preconditions for the test case to run correctly. (I'm not sure which case is here).

For example, if a test is
```
TestDivision() {
a = getnumber();
b = getnumber()
ASSERT_NEQ(b, 0);
c = div(a,b);

EXPECT_EQ(c, a/b);
}
```

Then the first assert should not be an EXPECT because it's a precondition check.

Another comment (related to the above): if an EXPECT fails, the test does not stop. But if an ASSERT fails, it does.

Another comment (related): ASSERTs on subroutines/helper functions do not work as expected. Some extra work is needed: https://github.com/google/googletest/blob/main/docs/advanced.md#asserting-on-subroutines

Reply:
@fcarreiro Thanks for the pointer. The check_circuit() is part of the test rather than a precondition check in this context. 

In the context of deterministic tests, I am not sure it matters so much. You will want to have everything green anyway. The ASSERT will allow you to save CPU time.